### PR TITLE
gcc@4.9: split args for macOS and Linux

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gcc@4.9.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class GccAT49 < Formula
   url "https://ftp.gnu.org/gnu/gcc/gcc-4.9.4/gcc-4.9.4.tar.bz2"
   mirror "https://ftpmirror.gnu.org/gcc/gcc-4.9.4/gcc-4.9.4.tar.bz2"
   sha256 "6c11d292cd01b294f9f84c9a59c230d80e9e4a47e5c6355f046bb36d4f358092"
-  revision 2
+  revision 3
```
Comment (carlocab) [inline Formula/gcc@4.9.rb:7]:
```suggestion
  revision 2
```
We can't rev bump this, as this only has a High Sierra bottle, and has a
```ruby
depends_on maximum_macos: [:high_sierra, :build]
```

**Comment by iMichka** [comment]
Right. I removed the rev-bump. Just FYI:

```
macOS
==> Analytics
install: 692 (30 days), 2,025 (90 days), 10,917 (365 days)
install-on-request: 682 (30 days), 2,006 (90 days), 9,839 (365 days)
build-error: 0 (30 days)

Linux
==> Analytics
install: 16 (30 days), 38 (90 days), 163 (365 days)
install-on-request: 16 (30 days), 38 (90 days), 163 (365 days)
build-error: 0 (30 days)
```

I think we can remove that formula in the next months, it's pretty EOL now.

**Comment by carlocab** [comment]
Let's deprecate it now then.

**Comment by iMichka** [comment]
Done. I used today's date as I could not find the real deprecation date.

**Review by carlocab — APPROVED** [review]

**Pull request merged**